src/index.js

```diff
@@ -50,7 +51,7 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // Avoid dispatching an action if the store is already up-to-date,
     // even if `history` wouldn't do anything if the location is the same
     if(getRouterState().path !== locationToString(location)) {
-      store.dispatch(updatePath(locationToString(location)));
+      store.dispatch(updatePath(locationToString(location), true));
```

Comment:
Won't `getRouterState().path` and `locationToString(location)` be equal here then if the transition already happened? What's the difference between this workflow and the normal one where you update the store which triggers a router update? The only difference is that this `history.listen` event will be delayed, but it should end up in the same place: the values are equal here.

Just trying to fully understand what I'm merging in.